src/Ormolu/Printer/Operators.hs

```diff
@@ -1,182 +1,365 @@
-{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiWayIf #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
 -- | This module helps handle operator chains composed of different
 -- operators that may have different precedence and fixities.
 module Ormolu.Printer.Operators
   ( OpTree (..),
+    OpInfo (..),
     opTreeLoc,
     reassociateOpTree,
+    isHardSplitterOp,
   )
 where
 
-import Data.Function (on)
-import qualified Data.List as L
-import Data.Map.Strict (Map)
-import qualified Data.Map.Strict as M
-import Data.Maybe (fromMaybe, mapMaybe)
-import GHC.Types.Fixity
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Map.Strict as Map
+import Data.Maybe (fromMaybe)
 import GHC.Types.Name.Occurrence (occNameString)
 import GHC.Types.Name.Reader
-import GHC.Types.SourceText
 import GHC.Types.SrcLoc
+import Ormolu.Fixity
 import Ormolu.Utils
 
--- | Intermediate representation of operator trees. It has two type
--- parameters: @ty@ is the type of sub-expressions, while @op@ is the type
--- of operators.
+-- | Intermediate representation of operator trees, where a branching is not
```

Comment:
Can you please go through the module and try to make sure that the haddocks are up to date with the current code?